pennylane/templates/layers/basic_entangler.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``BasicEntanglerLayers`` template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+from pennylane.ops import CNOT, RX
+from pennylane.templates import broadcast
+from pennylane.templates.utils import (
+    _check_shape,
+    _check_no_variable,
+    _check_wires,
+    _check_number_of_layers,
+    _get_shape,
+)
+
+
+@template
+def BasicEntanglerLayers(weights, wires, rotation=None):
+    r"""Layers consisting of one-parameter single-qubit rotations on each qubit, followed by a closed chain
+    or *ring* of CNOT gates.
+
+    The ring of CNOT gates connects every qubit with its neighbour,
+    with the last qubit being considered as a neighbour to the first qubit.
+
+    .. figure:: ../../_static/templates/layers/basic_entangler.png
+        :align: center
+        :width: 40%
+        :target: javascript:void(0);
+
+    The number of layers :math:`L` is determined by the first dimension of the argument ``weights``.
+    When using a single wire, the template only applies the single
+    qubit gates in each layer.
+
+    .. note::
+
+        This template follows the convention of dropping the entanglement between the last and the first
+        qubit when using only two wires, so the entangler is not repeated on the same wires.
+        In this case, only one CNOT gate is applied in each layer:
+
+        .. figure:: ../../_static/templates/layers/basic_entangler_2wires.png
+            :align: center
+            :width: 30%
+            :target: javascript:void(0);
+
+    Args:
+
+        weights (array[float]): array of weights with shape ``(L, len(wires))``, each weight is used as a parameter for the rotation
+        wires (Sequence[int] or int): qubit indices that the template acts on
+        rotation (pennylane.ops.Operation): one-parameter single-qubit gate to use,
+                                            if ``None``, :class:`~pennylane.ops.RX` is used as default
+    Raises:
+        ValueError: if inputs do not have the correct format
+
+    .. UsageDetails::
+
+        The template is used inside a qnode:
```

Comment:
```suggestion
        The template is used inside a :class:`~.BaseQNode`:
```

Reply:
Question: do we want to link to `BaseQNode` here, or the `QNode` constructor? The second is more userfacing, and the docstring makes more sense for users, even thought the first is more accurate.

Reply:
To be honest, I was also not sure about this. I'm happy if the `QNode` constructor is referenced at these places. One thing that would be nice to update is the following link (to `BaseQNode` I guess): https://github.com/XanaduAI/pennylane/blob/a76f103695e15668c22005a86a6d8e31d9b3c236/pennylane/qnodes/decorator.py#L35

Users could then quickly surf through this function, be directed to `BaseQNode` find the other variety of qnodes as well.

This change of the link would be out of scope for sure. :)

Reply:
Agree, the constructor should not be linking to itself in the docstring! But perhaps also add a warning in the BaseQNode (and other QNodes), that the are for advanced users, and the constructor should generally be used.